packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -156,19 +164,57 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
             },
           ],
         };
-      } else if (
+      }
+
+      const mappingsAreUnchanged =
+        // the source exists
+        Boolean(indices[source!]?.mappings?._meta?.migrationMappingPropertyHashes) &&
+        // ...and mappings are unchanged
+        !diffMappings(
+          /* actual */
+          indices[source!].mappings,
+          /* expected */
+          stateP.targetIndexMappings
+        );
+
+      if (mappingsAreUnchanged) {
+        const targetIndex = source!;
+
+        return {
+          ...stateP,
+          // We set a special flag that indicates we DO NOT need to reindex
+          controlState: 'OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT',
+          sourceIndex: Option.none,
+          targetIndex,
+          sourceIndexMappings: indices[source!].mappings,
+          // Setting this to "undefined" will trigger UPDATE_TARGET_MAPPINGS
+          // which we always want to do for new versions
+          targetIndexCurrentMappings: undefined,
+          targetIndexMappings: mergeMigrationMappingPropertyHashes(
+            stateP.targetIndexMappings,
+            indices[source!].mappings
+          ),
```

Comment:
Is this `mergeMigrationMappingPropertyHashes` step needed here? I couldn't see a difference in behaviour with or without it?

Reply:
Need to double check this, but IIRC `diffMappings` ignores fields in `actual` that's not in `expected`. So the actual index might have mappings for a plugin that we don't know about and diffMappings will say: "these mappings are the same 😄 ".

Because we're not removing the mappings of the unknown plugin, we shouldn't remove the _meta for that plugin's types either which is what mergMigrationMappingPropertyHashes accomplishes.